internal/armada/repository/job_test.go

```diff
@@ -14,6 +14,14 @@ import (
 	"github.com/G-Research/armada/pkg/api"
 )
 
+func TestJobDoubleSubmit(t *testing.T) {
+	withRepository(func(r *RedisJobRepository) {
+		job1 := addTestJobWithClientId(t, r, "queue1", "my-job-1")
+		job2 := addTestJobWithClientId(t, r, "queue1", "my-job-1")
+		assert.Equal(t, job1.Id, job2.Id)
+	})
+}
+
```

Comment:
We don't seem to have a test for "two separate queues can have the same job ID"?